config/api/src/main/java/org/jboss/windup/config/AbstractRuleLifecycleListener.java

```diff
@@ -6,26 +6,29 @@
 /**
  * This provides a set of default (empty) methods that make it easy to implement {@link RuleLifecycleListener}s that only need to override a subset of
  * the available methods.
- * 
- * @author <a href="mailto:jesse.sightler@gmail.com">Jesse Sightler</a>
  *
+ * @author <a href="mailto:jesse.sightler@gmail.com">Jesse Sightler</a>
+ * @author <a href="mailto:zizka@seznam.cz">Ondrej Zizka</a>
  */
 public abstract class AbstractRuleLifecycleListener implements RuleLifecycleListener
 {
-
     @Override
     public void beforeExecution(GraphRewrite event)
     {
     }
 
     @Override
-    public void beforeRuleEvaluation(GraphRewrite event, Rule rule, EvaluationContext context)
+    public boolean beforeRuleEvaluation(GraphRewrite event, Rule rule, EvaluationContext context)
```

Comment:
I think that it would be simpler to add a "isCancelled" flag rather than have every method return a boolean. It feels like it is overloading the functionality of these listener methods to add this as a return value here.